src/hooks/react/useDebounceState.ts

```diff
@@ -49,8 +60,28 @@ export const useDebounceState = ({ initialValue, wait = 500, transformCallback,
   );
 
   useEffect(() => {
-    previousDebounced !== debounced && dispatchCallback && dispatchCallback(debounced);
-  }, [debounced, dispatchCallback, previousDebounced]);
+    if (shouldDispatch && previousDebounced !== debounced && dispatchCallback) {
+      dispatchCallback(debounced);
+    }
+  }, [debounced, dispatchCallback, previousDebounced, shouldDispatch]);
+
+  // Mechanism to reset local state input when the transaction is executed.
+  // In case no reset parameter is set to the hook, this process will not execute.
+  useEffect(() => {
+    if (value && debounced && !toReset && reset && previousReset !== reset) {
+      setToReset(true);
+      setShouldDispatch(false);
+    }
+    if (toReset && reset) {
+      setValue('');
+      setDebounced(null);
+      setToReset(false);
+    }
+
+    if (!shouldDispatch && !reset && previousReset !== reset) {
+      setShouldDispatch(true);
+    }
+  }, [reset, previousReset, setValue, value, debounced, toReset, shouldDispatch]);
```

Comment:
I am not sure I understand the reasoning behind the existence of `toReset` and `previousReset`. I think the code can be much more simplified with the one below:
```suggestion
  useEffect(() => {
    if (reset) {
      if (value && debounced) {
        setShouldDispatch(false);
      }
      setValue('');
      setDebounced(null);
    }

    if (!shouldDispatch && !reset) {
      setShouldDispatch(true);
    }
  }, [reset, setValue, value, debounced, shouldDispatch]);
```

I am sure I am missing some reasoning here - but its a good conversation starter

Reply:
Yeah, I concur. The whole `setToReset` seems very unncessary. We should basically only check that `reset && !prevReset === true` and then run the clean up logic (i.e. the `reset` value changed from `false -> true`).

Reply:
previous reset is not needed as we discussed. I have improved the code based on your suggestion.

Reply:
guys i will resolve the conversation because i will base the process on listening for transaction.id changes, so this discussion gets outdated.